internal/service/cloudbackupsnapshot/resource_cloud_backup_snapshot_test.go

```diff
@@ -110,12 +109,14 @@ func checkExists(resourceName string) resource.TestCheckFunc {
 		if ids["snapshot_id"] == "" {
 			return fmt.Errorf("no ID is set")
 		}
-		_, _, err := acc.ConnV2().CloudBackupsApi.GetReplicaSetBackup(context.Background(), ids["project_id"], ids["cluster_name"], ids["snapshot_id"]).Execute()
-		if err == nil {
-			return nil
+		if _, _, err := acc.ConnV2().CloudBackupsApi.GetReplicaSetBackup(context.Background(), ids["project_id"], ids["cluster_name"], ids["snapshot_id"]).Execute(); err != nil {
+			return fmt.Errorf("cloudBackupSnapshot (%s) does not exist", rs.Primary.Attributes["snapshot_id"])
 		}
-
-		return fmt.Errorf("cloudBackupSnapshot (%s) does not exist", rs.Primary.Attributes["snapshot_id"])
+		// needed as first call to cluster with new API will fail due to transition to ISS feature flag
+		if err := acc.CheckClusterExistsHandlingRetry(ids["project_id"], ids["cluster_name"]); err != nil {
```

Comment:
adding the check of the cluster so that failure in first call to the API is handled here, and not when verifying non empty plan in second step of migration test.